Quarkus 3.32.1 - quarkus:dependency-tree fails with "Failed to locate io.quarkus:quarkus-ide-launcher::jar among application dependencies"

### Describe the bug

Quarkus 3.32.1 - quarkus:dependency-tree fails with "Failed to locate io.quarkus:quarkus-ide-launcher::jar among application dependencies"

```
[ERROR] Failed to execute goal io.quarkus.platform:quarkus-maven-plugin:3.32.1:dependency-tree (default-cli) on project code-with-quarkus:
  Failed to resolve application model org.acme:code-with-quarkus::pom:1.0.0-SNAPSHOT dependencies: 
  Failed to locate io.quarkus:quarkus-ide-launcher::jar among application dependencies -> [Help 1]
```

Detailed stacktrace from `mvn -e quarkus:dependency-tree`
```
[ERROR] Failed to execute goal io.quarkus.platform:quarkus-maven-plugin:3.32.1:dependency-tree (default-cli) on project code-with-quarkus: Failed to resolve application model org.acme:code-with-quarkus::pom:1.0.0-SNAPSHOT dependencies: Failed to locate io.quarkus:quarkus-ide-launcher::jar among application dependencies -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal io.quarkus.platform:quarkus-maven-plugin:3.32.1:dependency-tree (default-cli) on project code-with-quarkus: Failed to resolve application model org.acme:code-with-quarkus::pom:1.0.0-SNAPSHOT dependencies
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:333)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:316)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:212)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:174)
    at org.apache.maven.lifecycle.internal.MojoExecutor.access$000 (MojoExecutor.java:75)
    at org.apache.maven.lifecycle.internal.MojoExecutor$1.run (MojoExecutor.java:162)
    at org.apache.maven.plugin.DefaultMojosExecutionStrategy.execute (DefaultMojosExecutionStrategy.java:39)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:159)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:105)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:73)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:53)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:118)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:261)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:173)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:101)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:919)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:285)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:207)
    at jdk.internal.reflect.DirectMethodHandleAccessor.invoke (DirectMethodHandleAccessor.java:104)
    at java.lang.reflect.Method.invoke (Method.java:565)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:255)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:201)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:361)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:314)
Caused by: org.apache.maven.plugin.MojoExecutionException: Failed to resolve application model org.acme:code-with-quarkus::pom:1.0.0-SNAPSHOT dependencies
    at io.quarkus.maven.DependencyTreeMojo.logTree (DependencyTreeMojo.java:164)
    at io.quarkus.maven.DependencyTreeMojo.execute (DependencyTreeMojo.java:123)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:126)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:328)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:316)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:212)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:174)
    at org.apache.maven.lifecycle.internal.MojoExecutor.access$000 (MojoExecutor.java:75)
    at org.apache.maven.lifecycle.internal.MojoExecutor$1.run (MojoExecutor.java:162)
    at org.apache.maven.plugin.DefaultMojosExecutionStrategy.execute (DefaultMojosExecutionStrategy.java:39)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:159)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:105)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:73)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:53)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:118)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:261)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:173)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:101)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:919)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:285)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:207)
    at jdk.internal.reflect.DirectMethodHandleAccessor.invoke (DirectMethodHandleAccessor.java:104)
    at java.lang.reflect.Method.invoke (Method.java:565)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:255)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:201)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:361)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:314)
Caused by: java.lang.IllegalArgumentException: Failed to locate io.quarkus:quarkus-ide-launcher::jar among application dependencies
    at io.quarkus.bootstrap.resolver.maven.ApplicationDependencyResolver$AppDepLogger.getResolvedDependency (ApplicationDependencyResolver.java:1312)
    at io.quarkus.bootstrap.resolver.maven.ApplicationDependencyResolver$AppDepLogger.logInternal (ApplicationDependencyResolver.java:1225)
    at io.quarkus.bootstrap.resolver.maven.ApplicationDependencyResolver$AppDepLogger.log (ApplicationDependencyResolver.java:1186)
    at io.quarkus.bootstrap.resolver.maven.ApplicationDependencyResolver$AppDepLogger.log (ApplicationDependencyResolver.java:1196)
    at io.quarkus.bootstrap.resolver.maven.ApplicationDependencyResolver$AppDepLogger.log (ApplicationDependencyResolver.java:1196)
    at io.quarkus.bootstrap.resolver.maven.ApplicationDependencyResolver$AppDepLogger.log (ApplicationDependencyResolver.java:1196)
    at io.quarkus.bootstrap.resolver.maven.ApplicationDependencyResolver$AppDepLogger.log (ApplicationDependencyResolver.java:1196)
    at io.quarkus.bootstrap.resolver.maven.ApplicationDependencyResolver.populateModelBuilder (ApplicationDependencyResolver.java:330)
    at io.quarkus.bootstrap.resolver.maven.ApplicationDependencyResolver.resolve (ApplicationDependencyResolver.java:222)
    at io.quarkus.bootstrap.resolver.BootstrapAppModelResolver.buildAppModel (BootstrapAppModelResolver.java:408)
    at io.quarkus.bootstrap.resolver.BootstrapAppModelResolver.doResolveModel (BootstrapAppModelResolver.java:340)
    at io.quarkus.bootstrap.resolver.BootstrapAppModelResolver.resolveManagedModel (BootstrapAppModelResolver.java:238)
    at io.quarkus.bootstrap.resolver.BootstrapAppModelResolver.resolveModel (BootstrapAppModelResolver.java:222)
    at io.quarkus.maven.DependencyTreeMojo.logTree (DependencyTreeMojo.java:162)
    at io.quarkus.maven.DependencyTreeMojo.execute (DependencyTreeMojo.java:123)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:126)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:328)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:316)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:212)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:174)
    at org.apache.maven.lifecycle.internal.MojoExecutor.access$000 (MojoExecutor.java:75)
    at org.apache.maven.lifecycle.internal.MojoExecutor$1.run (MojoExecutor.java:162)
    at org.apache.maven.plugin.DefaultMojosExecutionStrategy.execute (DefaultMojosExecutionStrategy.java:39)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:159)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:105)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:73)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:53)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:118)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:261)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:173)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:101)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:919)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:285)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:207)
    at jdk.internal.reflect.DirectMethodHandleAccessor.invoke (DirectMethodHandleAccessor.java:104)
    at java.lang.reflect.Method.invoke (Method.java:565)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:255)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:201)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:361)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:314)
```

### Expected behavior

command ends without any error

### Actual behavior

command ends with error

### How to Reproduce?

* Get application from https://code.quarkus.io/
* Run `mvn -e quarkus:dependency-tree`

### Output of `uname -a` or `ver`

macOS 26.3

### Output of `java -version`

Java 21

### Quarkus version or git rev

3.32.1

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

--- a/devtools/maven/src/test/java/io/quarkus/maven/BasicDependencyTreeTestBase.java
+++ b/devtools/maven/src/test/java/io/quarkus/maven/BasicDependencyTreeTestBase.java
@@ -11,6 +11,8 @@ abstract class BasicDependencyTreeTestBase extends DependencyTreeMojoTestBase {
 
         final TsQuarkusExt coreExt = new TsQuarkusExt("test-core-ext");
         app = TsArtifact.jar("test-app")
+                .addDependency(TsArtifact.jar("io.quarkus", "quarkus-core", "1.0.0")
+                        .addDependency(TsArtifact.jar("io.quarkus", "quarkus-ide-launcher", "1.0.0")))
                 .addDependency(new TsArtifact(TsArtifact.DEFAULT_GROUP_ID, "artifact-with-classifier", "classifier", "jar",
                         TsArtifact.DEFAULT_VERSION))
                 .addDependency(new TsQuarkusExt("test-ext2")

--- a/devtools/maven/src/test/resources/test-app-1.jar.dev
+++ b/devtools/maven/src/test/resources/test-app-1.jar.dev
@@ -1,5 +1,6 @@
 [info] Quarkus application DEV mode build dependency tree:
 [info] io.quarkus.bootstrap.test:test-app:pom:1
+[info] ├─ io.quarkus:quarkus-core:1.0.0 (compile)
 [info] ├─ io.quarkus.bootstrap.test:artifact-with-classifier:classifier:1 (compile)
 [info] ├─ io.quarkus.bootstrap.test:test-ext2-deployment:1 (compile)
 [info] │  ├─ io.quarkus.bootstrap.test:test-ext2:1 (compile)

--- a/devtools/maven/src/test/resources/test-app-1.jar.prod
+++ b/devtools/maven/src/test/resources/test-app-1.jar.prod
@@ -1,5 +1,6 @@
 [info] Quarkus application PROD mode build dependency tree:
 [info] io.quarkus.bootstrap.test:test-app:pom:1
+[info] ├─ io.quarkus:quarkus-core:1.0.0 (compile)
 [info] ├─ io.quarkus.bootstrap.test:artifact-with-classifier:classifier:1 (compile)
 [info] ├─ io.quarkus.bootstrap.test:test-ext2-deployment:1 (compile)
 [info] │  ├─ io.quarkus.bootstrap.test:test-ext2:1 (compile)

--- a/devtools/maven/src/test/resources/test-app-1.jar.test
+++ b/devtools/maven/src/test/resources/test-app-1.jar.test
@@ -1,5 +1,6 @@
 [info] Quarkus application TEST mode build dependency tree:
 [info] io.quarkus.bootstrap.test:test-app:pom:1
+[info] ├─ io.quarkus:quarkus-core:1.0.0 (compile)
 [info] ├─ io.quarkus.bootstrap.test:artifact-with-classifier:classifier:1 (compile)
 [info] ├─ io.quarkus.bootstrap.test:test-ext2-deployment:1 (compile)
 [info] │  ├─ io.quarkus.bootstrap.test:test-ext2:1 (compile)

--- a/independent-projects/bootstrap/maven-resolver/src/main/java/io/quarkus/bootstrap/resolver/maven/ApplicationDependencyResolver.java
+++ b/independent-projects/bootstrap/maven-resolver/src/main/java/io/quarkus/bootstrap/resolver/maven/ApplicationDependencyResolver.java
@@ -292,6 +292,7 @@ public class ApplicationDependencyResolver {
      */
     private void populateModelBuilder(DependencyNode root) {
         var app = new AppDep(root);
+        app.resolvedDep = appBuilder.getApplicationArtifact();
         initMissingDependencies(app);
         for (var d : app.children) {
             d.addToModel();
@@ -1178,16 +1179,17 @@ public class ApplicationDependencyResolver {
 
         void log(AppDep dep) {
             logInternal(dep);
-            final int childrenTotal = dep.node.getChildren().size();
-            if (childrenTotal > 0) {
-                if (childrenTotal == 1) {
+            List<AppDep> children = getChildrenToLog(dep);
+            if (!children.isEmpty()) {
+                if (children.size() == 1) {
                     depth.add(false);
-                    log(dep.children.get(0));
+                    log(children.get(0));
                 } else {
                     depth.add(true);
+                    int childrenTotal = children.size();
                     int i = 0;
                     while (i < childrenTotal) {
-                        log(dep.children.get(i++));
+                        log(children.get(i++));
                         if (i == childrenTotal - 1) {
                             depth.set(depth.size() - 1, false);
                         }
@@ -1195,6 +1197,26 @@ public class ApplicationDependencyResolver {
                 }
                 depth.remove(depth.size() - 1);
             }
+        }
+
+        private List<AppDep> getChildrenToLog(AppDep dep) {
+            if (dep.resolvedDep.getDependencies().isEmpty() || dep.children.isEmpty()) {
+                return List.of();
+            }
+            if (dep.children.size() == 1) {
+                // some dependencies could be filtered out by applying user-configured exclusions
+                if (dep.resolvedDep.getDependencies().contains(dep.children.get(0).resolvedDep.getArtifactCoords())) {
+                    return dep.children;
+                }
+                return List.of();
+            }
+            final List<AppDep> children = new ArrayList<>(dep.children.size());
+            for (var child : dep.children) {
+                if (dep.resolvedDep.getDependencies().contains(child.resolvedDep.getArtifactCoords())) {
+                    children.add(child);
+                }
+            }
+            return children;
         }
 
         private void logInternal(AppDep dep) {
@@ -1216,7 +1238,7 @@ public class ApplicationDependencyResolver {
                     buf.append('\u2514').append('\u2500').append(' ');
                 }
             }
-            var resolvedDep = getResolvedDependency(getKey(dep.node.getArtifact()));
+            final ResolvedDependencyBuilder resolvedDep = dep.resolvedDep;
             buf.append(resolvedDep.toCompactCoords());
             if (!depth.isEmpty()) {
                 appendFlags(buf, resolvedDep);

